api/base/pagination.py

```diff
@@ -175,3 +178,131 @@ def get_paginated_response(self, data):
         total_bibliographic = len(node.visible_contributor_ids)
         response_dict['links']['meta']['total_bibliographic'] = total_bibliographic
         return Response(response_dict)
+
+
+class SearchPaginator(DjangoPaginator):
+
+    def __init__(self, object_list, per_page):
+        super(SearchPaginator, self).__init__(object_list, per_page)
+
+    def load_obj(self, obj_id, obj_type):
+        if obj_type in ['project', 'component', 'registration']:
```

Comment:
I wonder if it's worth having a "search type to model" function that makes this generic. As we add more search types (preprints, meetings) it would be nice to have it centralized.